runtime/src/main/java/org/corfudb/runtime/view/ManagementView.java

```diff
@@ -221,6 +226,86 @@ private ClusterStatus getClusterHealth(Layout layout, Set<String> peerResponsive
         return peerResponsiveNodes;
     }
 
+    /**
+     * Fetches the initial layout to fetch the cluster status.
+     * This does not consume the corfu runtime layout as that can get stuck in an indefinite retry
+     * loop.
+     *
+     * @param layoutServers Layout Servers to fetch the layout from.
+     * @return Latest layout from the servers or null if none of them responded with a layout.
+     */
+    private Layout getClusterStatusLayout(List<String> layoutServers) {
+        Layout layout = null;
+        Map<String, CompletableFuture<Layout>> layoutFuturesMap = new HashMap<>();
+        // Get layout futures for layout requests to all layout servers.
+        for (String server : layoutServers) {
+            try {
+                IClientRouter router = runtime.getRouter(server);
+                layoutFuturesMap.put(server,
+                        new LayoutClient(router, Layout.INVALID_EPOCH).getLayout());
+            } catch (Exception e) {
```

Comment:
Why catch Exception, I think we should only expect a NetworkException/TimeoutException here since it will call the channel directly. Make the exception specific.

Reply:
Yes only NetworkException while creating the router. This does not wait for the callback.
Done.